src/Altinn.App.Core/Features/Action/SigningUserAction.cs

```diff
@@ -92,17 +100,39 @@ public async Task<UserActionResult> HandleAction(UserActionContext context)
         );
     }
 
+    private static bool ShouldSign(
+        ProcessTask currentTask,
+        List<DataElement> dataElements,
+        List<DataType>? dataTypesToSign
+    )
+    {
+        var signatureIsConfigured =
+            currentTask.ExtensionElements?.TaskExtension?.SignatureConfiguration?.SignatureDataType is not null;
+        if (dataTypesToSign.IsNullOrEmpty() || !signatureIsConfigured)
```

Comment:
Is the `IsNullOrEmpty` function worth it in this PR? when we have concrete type like a `List<>` I think we might as well use `dataTypesToSign is null or []` - codegen for that is a null check and `Count` check

Reply:
I saw that .IsNullOrEmpty was used elsewhere as well (PdfService) - by using Microsoft.IdentityModel.Tokens, line 73/74.

Reply:
Looking at it again, I think the implementation in PdfService was looking to use string.IsNullOrEmpty(), and instead did stringvar.IsNullOrEmpty by accident. With this in mind, it may not be worth it